go/keyset/manager.go

```diff
@@ -64,12 +90,117 @@ func (km *Manager) Rotate(kt *tinkpb.KeyTemplate) error {
 		KeyId:            keyID,
 		OutputPrefixType: kt.OutputPrefixType,
 	}
-	// Set the new key as the primary key
 	km.ks.Key = append(km.ks.Key, key)
-	km.ks.PrimaryKeyId = keyID
+	return keyID, nil
+}
+
+// SetPrimary sets the key with given keyID as primary
+// returns an error if the key is not found or not enabled
+func (km *Manager) SetPrimary(keyID uint32) error {
+	if km.ks == nil {
+		return fmt.Errorf("keyset_manager: cannot set primary, no keyset")
+	}
+	for i := 0; i < len(km.ks.Key); i++ {
+		if km.ks.Key[i].KeyId == keyID {
+			if km.ks.Key[i].Status == tinkpb.KeyStatusType_ENABLED {
+				km.ks.PrimaryKeyId = keyID
+				return nil
+			}
+			return fmt.Errorf("keyset_manager: cannot set key as primary because it's not enabled")
+		}
+	}
+	return fmt.Errorf("keyset_manager: key with id %d not found", keyID)
+}
+
+// Enable will enable the key with given keyID
+// returns an error if the key is not found or is not enabled or disabled already
+func (km *Manager) Enable(keyID uint32) error {
+	if km.ks == nil {
+		return fmt.Errorf("keyset_manager: cannot enable key, no keyset")
+	}
+	for i := 0; i < len(km.ks.Key); i++ {
+		if km.ks.Key[i].KeyId == keyID {
+			if km.ks.Key[i].Status == tinkpb.KeyStatusType_ENABLED || km.ks.Key[i].Status == tinkpb.KeyStatusType_DISABLED {
+				km.ks.Key[i].Status = tinkpb.KeyStatusType_ENABLED
+				return nil
+			}
+			return fmt.Errorf("keyset_manager: cannot enable key with id %d with status %s", keyID, km.ks.Key[i].Status.String())
+		}
+	}
+	return fmt.Errorf("keyset_manager: key with id %d not found", keyID)
+}
+
+// Disable will disable the key with given keyID
+// returns an error if the key is not found or it is the primary key
+func (km *Manager) Disable(keyID uint32) error {
+	if km.ks == nil {
+		return fmt.Errorf("keyset_manager: cannot disable key, no keyset")
+	}
+	if km.ks.PrimaryKeyId == keyID {
+		return fmt.Errorf("keyset_manager: cannot disable the primary key")
+	}
+	for i := 0; i < len(km.ks.Key); i++ {
+		if km.ks.Key[i].KeyId == keyID {
+			if km.ks.Key[i].Status == tinkpb.KeyStatusType_ENABLED || km.ks.Key[i].Status == tinkpb.KeyStatusType_DISABLED {
+				km.ks.Key[i].Status = tinkpb.KeyStatusType_DISABLED
+				return nil
+			}
+			return fmt.Errorf("keyset_manager: cannot disable key with id %d with status %s", keyID, km.ks.Key[i].Status.String())
+		}
+	}
+	return fmt.Errorf("keyset_manager: key with id %d not found", keyID)
+}
+
+// Delete will delete the key with given keyID, removing the key from the keyset entirely
+// returns an error if the key is not found or it is the primary key
+func (km *Manager) Delete(keyID uint32) error {
```

Comment:
thoughts on not exposing this? It's available in Java, but perhaps doesn't need to be added here. Just seems dangerous to expose.

Reply:
As we're going to add only enough to unblock you, if you don't need this, it's okay not to expose it.